docs/contributing/contribution-checklists.md

```diff
@@ -281,80 +284,139 @@ More details about this code generation, including fixes for potential error mes
 
   ```go
+  // Typically declared near conn := /* ... */
+  defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+  tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
+  
   input := &ec2.CreateFleetInput{
     /* ... other configuration ... */
-    TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeFleet),
+    TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeFleet),
   }
   ```
 
 - In the resource `Read` function, implement the logic to convert the service tags to save them into the Terraform state for drift detection, e.g. with EKS Clusters (which had the tags available in the DescribeCluster API call):
 
   ```go
   // Typically declared near conn := /* ... */
+  defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
   ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+  
+  /* ... other d.Set(...) logic ... */
 
-  if err := d.Set("tags", keyvaluetags.EksKeyValueTags(cluster.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+  tags := keyvaluetags.EksKeyValueTags(cluster.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+  
+  if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
     return fmt.Errorf("error setting tags: %w", err)
   }
+  
+  if err := d.Set("tags_all", tags.Map()); err != nil {
+    return fmt.Errorf("error setting tags_all: %w", err)
+  }
   ```
 
   If the service API does not return the tags directly from reading the resource and requires a separate API call, its possible to use the `keyvaluetags` functionality like the following, e.g. with Athena Workgroups:
 
   ```go
   // Typically declared near conn := /* ... */
+  defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
   ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
+  /* ... other d.Set(...) logic ... */
+  
   tags, err := keyvaluetags.AthenaListTags(conn, arn.String())
 
   if err != nil {
     return fmt.Errorf("error listing tags for resource (%s): %s", arn, err)
   }
 
-  if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+  tags = tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+  
+  if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
     return fmt.Errorf("error setting tags: %w", err)
   }
+  
+  if err := d.Set("tags_all", tags.Map()); err != nil {
+    return fmt.Errorf("error setting tags_all: %w", err)
+  }
   ```
 
 - In the resource `Update` function (this may be the first functionality requiring the creation of the `Update` function), implement the logic to handle tagging updates, e.g. with EKS Clusters:
 
   ```go
-  if d.HasChange("tags") {
-    o, n := d.GetChange("tags")
+  if d.HasChange("tags_all") {
+    o, n := d.GetChange("tags_all")
```

Comment:
Do we want a note on having to use `d.HasChangesExcept("tags", "tags_all")` if currently using `d.HasChangesExcept("tags")`?

Reply:
definitely 👍 will add!

Reply:
added in https://github.com/hashicorp/terraform-provider-aws/pull/19084/commits/a5e338530f680f2b28a274ae6c1e2c987ca34206